src/servers/src/postgres/handler.rs

```diff
@@ -124,8 +146,34 @@ where
         .map(move |row| {
             row.and_then(|row| {
                 let mut encoder = DataRowEncoder::new(pg_schema_ref.clone());
-                for value in row.iter() {
-                    encode_value(value, &mut encoder)?;
+                for (idx, value) in row.iter().enumerate() {
+                    match value {
```

Comment:
Can we move all these check into `encode_value`? Also we can pass query_ctx into `encode_value` to get configuration.